src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```diff
@@ -767,6 +829,141 @@ private static ISymbol SymbolFromArrayCreationOperation(IArrayCreationOperation
             return false;
         }
 
+#pragma warning disable CA1054,CA1055 // url should be of type URI
+        private static string? GetMessageAndURLFromAttributeConstructor(AttributeData attribute, out string? url)
+#pragma warning restore CA1054,CA1055
+        {
+            string? message = null;
+            url = null;
+            ImmutableArray<TypedConstant> constructorArguments = attribute.ConstructorArguments;
+            if (constructorArguments.Length != 0)
+            {
+                if (constructorArguments.First().Value is string messageValue)
+                {
+                    message = messageValue;
+                }
+            }
+
+            ImmutableArray<System.Collections.Generic.KeyValuePair<string, TypedConstant>> namedArguments = attribute.NamedArguments;
+            if (namedArguments.Length != 0)
+            {
+                foreach (System.Collections.Generic.KeyValuePair<string, TypedConstant> namedArgument in namedArguments)
+                {
+                    if (namedArgument.Key == "Message")
```

Comment:
I don't think this needs symbol level checks, but if there's a valid reason to do so, please let me know